### ISSUE TITLE
Future rejection should also reject children futures

### ISSUE THREAD

**Issue by othierry** [othierry/future#6]
When a future rejects, the children futures are not rejected. This causes `fail` blocks of the children future to never be called.

**Example**

Without children
```
Future<Int> {
  throw NSError(...)
}.fail { error in
  // This is called as expected
}
```

With children
```
Future<Int> {
  throw NSError(...)
}.then { x -> String in // Child future
  return "\(x)"
}.fail { error in
  // This is never called as fail block belongs to the child future
}
```

**Comment by othierry** [comment]
Fixed and test case added (The actual test case was wrong and made the error passing...).
Will roll out new version.

**Comment by othierry** [comment]
Fixed in `2.0.0rc3` and available on cocoapods.
